From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 18 Jul 2017 19:17:24 +0200
Subject: [PATCH] network: public: Use arm-virtual2 POD config

grep -e "10\.16\.0\." -R . -l | \
  xargs sed -i \
    -e 's/10\.16\.0\./10.0.9./g' \
    -e 's/10\.0\.9\.254/10.0.9.200/g'
    -e 's/10\.0\.9\.1/10.0.9.254/g'

NOTE: This should be converted into a dynamic configuration read from
the universal POD descriptor in securedlab, once that is ready.

Until then, just align the public network configuration used by the
virtual POD with the Enea lab configuration specific to arm-virtual2
(i.e. public network on 10.0.9.0/24).

NOTE: Replace the gateway at 10.16.0.1 (now 10.0.9.1) with the same
IP address as our lab's gateway (10.0.9.254), to keep both possible
network layouts in sync (using all virtual networks created via
virsh, respectively our lab's static config).
This will ensure deploys continue to work in both enviroments.

Also, since our new gateway resides at 10.0.9.254, trim the DHCP pool
range for the public network to not include that address.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/config/states/networks                                          | 2 +-
 mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/infra/config.yml  | 6 +++---
 .../classes/cluster/virtual-mcp-ocata-odl/openstack/init.yml        | 2 +-
 .../classes/cluster/virtual-mcp-ocata-ovs-dpdk/infra/config.yml     | 6 +++---
 .../classes/cluster/virtual-mcp-ocata-ovs-dpdk/openstack/init.yml   | 2 +-
 mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/infra/config.yml  | 6 +++---
 .../classes/cluster/virtual-mcp-ocata-ovs/openstack/init.yml        | 2 +-
 mcp/scripts/net_public.xml                                          | 2 +-
 8 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/mcp/config/states/networks b/mcp/config/states/networks
index 205e0a9..df4c0bb 100755
--- a/mcp/config/states/networks
+++ b/mcp/config/states/networks
@@ -1,3 +1,3 @@
 salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack compute service list; openstack network agent list; openstack stack list; openstack volume service list"
 salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack network create --external --default --provider-network-type flat --provider-physical-network physnet1 floating_net"
-salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack subnet create --gateway 10.16.0.1 --no-dhcp --allocation-pool start=10.16.0.130,end=10.16.0.254 --network floating_net --subnet-range 10.16.0.0/24 floating_subnet"
+salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack subnet create --gateway 10.0.9.254 --no-dhcp --allocation-pool start=10.0.9.130,end=10.0.9.200 --network floating_net --subnet-range 10.0.9.0/24 floating_subnet"
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/infra/config.yml
index 57c6cec..f519e22 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/infra/config.yml
@@ -51,16 +51,16 @@ parameters:
           params:
             single_address: 172.16.10.105
             tenant_address: 10.1.0.105
-            external_address: 10.16.0.105
+            external_address: 10.0.9.105
         openstack_compute_node02:
           params:
             single_address: 172.16.10.106
             tenant_address: 10.1.0.106
-            external_address: 10.16.0.106
+            external_address: 10.0.9.106
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
         opendaylight_control_node01:
           classes:
           - cluster.${_param:cluster_name}.opendaylight.control
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/openstack/init.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/openstack/init.yml
index 66e5a97..98625ff 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-odl/openstack/init.yml
@@ -6,7 +6,7 @@ parameters:
     openstack_region: RegionOne
     admin_email: root@localhost
     cluster_public_protocol: http
-    cluster_public_host: 10.16.0.101
+    cluster_public_host: 10.0.9.101
     neutron_public_protocol: http
     neutron_control_dvr: False
     neutron_tenant_network_types: "flat,vxlan"
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/infra/config.yml
index aec6cde..b69258f 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/infra/config.yml
@@ -50,17 +50,17 @@ parameters:
           params:
             single_address: 172.16.10.105
             tenant_address: 10.1.0.105
-            external_address: 10.16.0.105
+            external_address: 10.0.9.105
             dpdk0_name: enp3s0
             dpdk0_pci: '"0000:00:05.0"'
         openstack_compute_node02:
           params:
             single_address: 172.16.10.106
             tenant_address: 10.1.0.106
-            external_address: 10.16.0.106
+            external_address: 10.0.9.106
             dpdk0_name: enp3s0
             dpdk0_pci: '"0000:00:05.0"'
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/openstack/init.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/openstack/init.yml
index dfc0b51..b106e86 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs-dpdk/openstack/init.yml
@@ -6,7 +6,7 @@ parameters:
     openstack_region: RegionOne
     admin_email: root@localhost
     cluster_public_protocol: http
-    cluster_public_host: 10.16.0.101
+    cluster_public_host: 10.0.9.101
     neutron_public_protocol: http
     neutron_control_dvr: False
     neutron_tenant_network_types: "flat,vlan"
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/infra/config.yml
index 8cb0992..c8f8477 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/infra/config.yml
@@ -50,13 +50,13 @@ parameters:
           params:
             single_address: 172.16.10.105
             tenant_address: 10.1.0.105
-            external_address: 10.16.0.105
+            external_address: 10.0.9.105
         openstack_compute_node02:
           params:
             single_address: 172.16.10.106
             tenant_address: 10.1.0.106
-            external_address: 10.16.0.106
+            external_address: 10.0.9.106
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/openstack/init.yml b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/openstack/init.yml
index 7079fd1..fccde1f 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-ocata-ovs/openstack/init.yml
@@ -6,7 +6,7 @@ parameters:
     openstack_region: RegionOne
     admin_email: root@localhost
     cluster_public_protocol: http
-    cluster_public_host: 10.16.0.101
+    cluster_public_host: 10.0.9.101
     neutron_public_protocol: http
     neutron_control_dvr: False
     neutron_tenant_network_types: "flat,vxlan"
diff --git a/mcp/scripts/net_public.xml b/mcp/scripts/net_public.xml
index 61650d5..f455480 100644
--- a/mcp/scripts/net_public.xml
+++ b/mcp/scripts/net_public.xml
@@ -2,5 +2,5 @@
   <name>public</name>
   <bridge name="public"/>
   <forward mode="nat"/>
-  <ip address="10.16.0.1" netmask="255.255.255.0" />
+  <ip address="10.0.9.254" netmask="255.255.255.0" />
 </network>
